modules/gapi/include/opencv2/gapi/sycl/gsyclkernel.hpp

```diff
@@ -0,0 +1,176 @@
+#ifndef OPENCV_GAPI_GSYCLKERNEL_HPP
+#define OPENCV_GAPI_GSYCLKERNEL_HPP
+
+#include <vector>
+#include <functional>
+#include <map>
+#include <unordered_map>
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/gapi/gcommon.hpp>
+#include <opencv2/gapi/gkernel.hpp>
+#include <opencv2/gapi/garg.hpp>
+
+// FIXME: These are not included with OpenCV by default
+// This is protected by HAVE_SYCL in gsyclbackend.hpp, is this included anywhere else?
+#include <CL/sycl.hpp>
+#include <CL/sycl/backend/opencl.hpp> 
```

Comment:
I think that we do, because there are sycl::buffers<> in function signatures below and these are sycl specific datatypes. I could be wrong though.

Reply:
Can they be forward declared? I would think so, since you only have references to SYCL types in this header. 
```
namespace sycl {
    template <typename T> class buffer;
    class queue;
    class context;
} // namespace sycl
```